packages/consent/consent-tools/src/types/wrapper.ts

```diff
@@ -0,0 +1,104 @@
+import type { CreateWrapperSettings } from './settings'
+
+export interface AnalyticsBrowserSettings {
+  writeKey: string
+  cdnURL?: string
+  cdnSettings?: CDNSettings
+}
+
+/**
+ * 2nd arg to AnalyticsBrowser.load / analytics
+ */
+export interface InitOptions {
+  updateCDNSettings(cdnSettings: CDNSettings): CDNSettings
+}
+
+/**
+ * This interface is a stub of the actual Segment analytics instance.
+ * This can be either:
+ * - window.analytics (i.e `AnalyticsSnippet`)
+ * - the instance returned by `AnalyticsBrowser.load({...})`
+ * - the instance created by `new AnalyticsBrowser(...)`
+ */
+export interface AnyAnalytics {
+  addSourceMiddleware(...args: any[]): any
+  on(event: 'initialize', callback: (settings: CDNSettings) => void): void
+
+  /**
+   * This interface is meant to be compatible with both the snippet (`window.analytics.load`)
+   * and the npm lib (`AnalyticsBrowser.load`)
+   */
+  load(
+    writeKeyOrSettings: AnalyticsBrowserSettings | string,
+    options?: InitOptions
+  ): void
+}
+
+/**
+ * This function returns a "wrapped" version of analytics.
+ */
+export interface Wrapper {
+  // Returns void rather than analytics to emphasize that this function replaces the .load function of the underlying instance.
+  (analytics: AnyAnalytics): void
+}
+
+/**
+ * This function returns a function which returns a "wrapped" version of analytics
+ */
+export interface CreateWrapper {
+  (options: CreateWrapperSettings): Wrapper
+}
+
+export interface Categories {
+  [category: string]: boolean
+}
+
+export interface IntegrationCategoryMappings {
+  [integrationName: string]: string[]
+}
+
+export interface CDNSettings {
+  integrations: CDNSettingsIntegrations
+  remotePlugins?: CDNSettingsRemotePlugin[]
+  consentSettings?: {
+    // all unique categories keys
+    allCategories: string[]
+  }
+}
+
+/**
+ *CDN Settings Integrations object.
+ * @example
+ * { "Fullstory": {...}, "Braze Web Mode (Actions)": {...}}
+ */
+export interface CDNSettingsIntegrations {
+  [integrationName: string]: { [key: string]: any }
+}
+
+export interface CDNSettingsRemotePlugin {
```

Comment:
It is actually required for type safety -- since we are modifying a list of plugins (that are types with the concrete type via analytics), Typescript will complain if we use a minimal interface ('could be instantiated with a different type' etc).